### PR TITLE
[ROCm] Support all ROCm supported uarchs simultaneously: gfx803, gfx900, gfx906

### PR DESCRIPTION
**Pull request by iotamudelta**
Correct misspelled flag.

Remove dependency on debug flag (HCC_AMDGPU_TARGET)

### CODE REVIEW

**Comment by iotamudelta** [comment]
@bddppq this is the follow-up to my comment in #17329 . I think this is more robust and also adds unconditional support for all ROCm uarchs, making user's lifes easier.

File: cmake/public/LoadHIP.cmake [hunk 1]
```diff
@@ -94,12 +94,6 @@ ELSE()
   SET(MIOPEN_PATH $ENV{MIOPEN_PATH})
 ENDIF()
 
-IF(NOT DEFINED ENV{HCC_AMDGPU_TARGET})
-  SET(HCC_AMDGPU_TARGET gfx900;gfx906)
-ELSE()
```
Comment (bddppq) [inline cmake/public/LoadHIP.cmake]: We can add gfx803 to the default list, but I would still prefer to keep this user-configurable.

**Comment by iotamudelta** [comment]
@bddppq sounds good. Can we use another environment variable though? `PYTORCH_ROCM_ARCHS` or something since the current one doubles as a hcc debug flag and with hip-clang incoming, I'd like to make this truly compiler independent.

**Comment by bddppq** [comment]
@iotamudelta Sure renaming it sounds good to me

**Review by bddppq — APPROVED** [review]

**Pull request closed without merging**